sotodlib/tod_ops/fft_ops.py

```diff
@@ -293,3 +294,115 @@ def calc_wn(aman, pxx=None, freqs=None, low_f=5, high_f=10):
+    signal : nparray
+        Signal sized ndets x nsamps to fit noise model to.
+        Default is None which corresponds to aman.signal.
+    f : nparray
+        Frequency of PSD of signal.
+        Default is None which calculates f, pxx from signal.
+    pxx : nparray
+        PSD sized ndets x len(f) which is fit to with model.
+        Default is None which calculates f, pxx from signal.
+    psdargs : dict
+        Dictionary of optional argument for ``scipy.signal.welch``
+    fwhite : tuple
+        Low and high frequency used to estimate white noise for initial
+        guess passed to ``scipy.signal.curve_fit``.
+    lowf : tuple
+        Frequency below which estimate of 1/f noise index and knee are estimated
+        for initial guess passed to ``scipy.signal.curve_fit``.
+    merge_fit : bool
+        Merges fit and fit statistics into input axis manager.
+    f_max : float
+        Maximum frequency to include in the fitting. This is particularly
+        important for lowpass filtered data such as that post demodulation
+        if the data is not downsampled after lowpass filtering.
+    merge_name : bool
+        If ``merge_fit`` is True then addes into axis manager with merge_name.
+    merge_psd : bool
+        If ``merg_psd`` is True then adds fres and Pxx to the axis manager.
+    Returns
+    -------
+    noise_fit_stats : AxisManager
+        If merge_fit is False then axis manager with fit and fit statistics
+        is returned otherwise nothing is returned and axis manager is wrapped
+        into input aman.
+    """
+    if signal is None:
+        signal = aman.signal
+
+    if f is None or pxx is None:
+        if psdargs is None:
+            f, pxx = calc_psd(aman, signal=signal, timestamps=aman.timestamps,
+                              merge=merge_psd)
+        else:
+            f, pxx = calc_psd(aman, signal=signal, timestamps=aman.timestamps,
+                              merge=merge_psd, **psdargs)
+    eix = np.argmin(np.abs(f-f_max))
+    f = f[1:eix]
+    pxx = pxx[:,1:eix]
+
+    fitout = np.zeros((aman.dets.count, 3))
+    # This is equal to np.sqrt(np.diag(cov)) when doing curve_fit
+    covout = np.zeros((aman.dets.count, 3))
+    for i in range(aman.dets.count):
+        p = pxx[i]
+        wnest = np.median(p[((f>fwhite[0]) & (f<fwhite[1]))])
+        pfit = np.polyfit(np.log10(f[f<lowf]), np.log10(p[f<lowf]), 1)
+        fidx = np.argmin(np.abs(10**np.polyval(pfit, np.log10(f)) - wnest))
+        p0 = [f[fidx], wnest, -pfit[0]]
+        res = minimize(neglnlike, p0, args=(f, p), method='Nelder-Mead')
+        try:
+            Hfun = ndt.Hessian(lambda params : neglnlike(params, f, p), full_output=True)
+            hessian_ndt, _ = Hfun(res['x'])
+            covout[i] = np.sqrt(np.diag(np.linalg.inv(hessian_ndt)))
+        except np.linalg.LinAlgError:
+            print(f'Cannot calculate Hessian for detector {aman.dets.vals[i]} skipping.')
+            covout[i] = [np.nan, np.nan, np.nan]
```

Comment:
So the `Nelder-Mead` method in minimize doesn't return a Hessian or covariance matrix. So I've imported this `numdifftools` package to calculate it. Is it ok to add a new dependency like this @mhasself or @kmharrington?

Reply:
I do hate adding packages.  But go for it.  Please store the full 3x3 covar though ... or don't call it covar.